File: src/leetCode/Ex692.java
package leetCode;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class Ex692 {
    public List<String> topKFrequent(String[] words, int k) {
        var map = new HashMap<String, Integer>();
        for (String word :
                words) {
            map.put(word, map.getOrDefault(word, 0) + 1);
        }
        var retList = new ArrayList<String>();
        retList.addAll(map.keySet());
        retList.sort((o1, o2) -> {
            if (map.get(o1).equals(map.get(o2))) {
                return o1.compareTo(o2);
            } else {
                return map.get(o2) - map.get(o1);
            }
        });
        return retList.subList(0, k);
    }
}

File: src/leetCode/Ex692.kt
package leetCode

fun topKFrequent(words:Array<String>, k:Int ):List<String> = HashMap<String, Int>().apply {
    words.forEach { this[it] = this.getOrDefault(it,0)+1 }
}.entries.sortedWith{ o1, o2->
    if (o1.value == o2.value) {
        o1.key.compareTo(o2.key)
    }else{
        o2.value-o1.value
    }
}.subList(0,k).map { it.key }
